src/Sylius/Bundle/ShopBundle/EventListener/OrderPromotionIntegrityChecker.php

```diff
@@ -65,7 +79,18 @@ public function check(ResourceControllerEvent $event): void
 
         Assert::isInstanceOf($order, OrderInterface::class);
 
-        $promotions = $order->getPromotions();
+        // we creae a new promotion collection and remove them from cart
+        // so we can verify with original conditions (without the price beeng applied befor check)
+
+        $promotions = new ArrayCollection($order->getPromotions()->toArray());
```

Comment:
I guess we don't need to put that array inside `ArrayCollection` again, `$order->getPromotions()->toArray()` would be enough.